packages/compass-query-bar/src/components/query-history/favorite-list.tsx

```diff
@@ -54,11 +63,16 @@ const FavoriteItem = ({
       onClick={onCardClick}
       data-testid="favorite-query-list-item"
       header={(isHovered: boolean) => (
-        <QueryItemHeading title={query._name} isHovered={isHovered}>
-          <CopyActionButton
-            onClick={() => copyToClipboard(formatQuery(attributes))}
-          />
-          <DeleteActionButton onClick={onDeleteClick} />
+        <QueryItemHeading title={query._name} isHovered={true}>
+          {isHovered && (
+            <CopyActionButton
+              onClick={() => copyToClipboard(formatQuery(attributes))}
+            />
+          )}
+          {isHovered && <DeleteActionButton onClick={onDeleteClick} />}
+          {isUpdateQuery && (
+            <OpenBulkUpdateActionButton onClick={onCardClick} />
+          )}
```

Comment:
Should this account for views (potentially other readonly collections)? We show `update` entries in the list, but selecting them doesn't open the update modal:

<img width="487" alt="image" src="https://github.com/mongodb-js/compass/assets/5036933/9ae57adc-7dd5-4e3e-ace5-afd9570f07c0">

Reply:
I don't understand the question, can you elaborate? Clicking on a favourite update opens the modal, unless you've found a bug, can you share how you reproduce it?

https://github.com/mongodb-js/compass/assets/1822138/fb5a5db7-d3fe-4f3a-946f-fdd4306ea3f7

Reply:
Sure! Here's one of the scenarios to repro this:

* Connect with a user that has readwrite privileges
* Save a bulkUpdate
* Disconnect and connect with a user that has readonly privileges or in a way that makes the connection readonly
* You see the bulkUpdate item in the drop down, but clicking it doesn't show update modal (probably because something correctly checking if readonly is true for either connection or namespace elsewhere in the plugin, but not here in any way)

Saved items here are filtered only by namespace, they are not attached to a specific connection (on purpose, you might want to use then on different connection, like dev and prod), so there is a decent chance that saved bulk update will show up in case where collection or cluster is readonly.

The same thing also applies to "My Queries" screen.

Should we account for that or is this expected behavior that we show the item that does nothing on click?

Reply:
This is a corner case that we didn't take into account when we design the feature, so probably we should discuss it further with Misba and Le Roux. I personally would grey-out update queries in readonly connections, because you might want still to read the query even if you can't run it.

Do you mind if I open a ticket so we can track this specific use case? I think it's important to get design input, because we might want to have another flow for readonly collections (IDK, a copy button instead of opening the modal?).

Reply:
Sure, if we address this before the feature is ready for public release, that's fine by me

Reply:
I created the ticket and mentioned you in the conversation in Slack. I'm resolving here so we can continue the discussion with Misba and Le Roux.